sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LoggingEventBuilder.java

```diff
@@ -126,7 +124,9 @@ public LoggingEventBuilder addKeyValue(String key, String value) {
      */
     public LoggingEventBuilder addKeyValue(String key, Object value) {
         if (this.isEnabled) {
-            addKeyValueInternal(key, value == null ? null : value.toString());
+            // Previously this eagerly called toString() on the value, but that can be expensive and unnecessary.
```

Comment:
no objections on doing `toString()` later, but the comment is not accurate - this is behind `isEnabled`, which is populated based on the log level

Reply:
True that this is behind the enable flag but if someone forgets to actually call log this does waste cycles.